src/main/java/org/snomed/aag/rest/AcceptanceController.java

```diff
@@ -44,26 +48,35 @@ public AcceptanceController(BranchSecurityService securityService,
             @ApiResponse(code = 403, message = "&bull; When the Branch cannot be found from the given branch path. <br/> &bull; When individual Criteria Items do not exist for the branch's expected acceptance criteria."),
             @ApiResponse(code = 200, message = "When the branch (or its parent) has acceptance criteria.", response = ProjectAcceptanceCriteriaDTO.class)
     })
-    @GetMapping("/{branch}")
-    public ResponseEntity<?> viewCriteriaItems(@ApiParam("The branch path.") @PathVariable(name = "branch") String branch) throws RestClientException {
-        branch = BranchPathUriUtil.decodePath(branch);
+	@GetMapping("/{branchPath}")
+	public ResponseEntity<?> viewCriteriaItems(@ApiParam("The branch path.") @PathVariable(name = "branchPath") String branchPath,
+											   @ApiParam("Toggle whether to only return criteria matching authoring flags stored on Branch.") @RequestParam(required = false) boolean matchAuthorFlags) throws RestClientException {
+		branchPath = BranchPathUriUtil.decodePath(branchPath);
+
+		// Check branch exists
+		Branch branch = securityService.getBranchOrThrow(branchPath);
+
+		//Find ProjectAcceptanceCriteria (including mandatory items)
+		ProjectAcceptanceCriteria projectAcceptanceCriteria = projectAcceptanceCriteriaService.findEffectiveCriteriaWithMandatoryItems(branchPath);
+		if (projectAcceptanceCriteria == null) {
+			throw new ServiceRuntimeException(String.format("Cannot find Acceptance Criteria for %s.", branchPath), HttpStatus.NOT_FOUND);
+		}
 
-        // Check branch exists
-        securityService.getBranchOrThrow(branch);
+		// Update complete flag for all Criteria Items on this branch
+		Set<CriteriaItem> items = projectAcceptanceCriteriaService.findItemsAndMarkSignOff(projectAcceptanceCriteria, branchPath);
 
-        //Find ProjectAcceptanceCriteria (including mandatory items)
-        ProjectAcceptanceCriteria projectAcceptanceCriteria = projectAcceptanceCriteriaService.findEffectiveCriteriaWithMandatoryItems(branch);
-        if (projectAcceptanceCriteria == null) {
-			throw new ServiceRuntimeException(String.format("Cannot find Acceptance Criteria for %s.", branch), HttpStatus.NOT_FOUND);
+		if (matchAuthorFlags) {
+			// Remove Criteria Items that do not have a matching flag in branch metadata
+			Set<String> authorFlags = getAuthorFlags(branch).keySet();
+			if (!authorFlags.isEmpty()) {
+				items.removeIf(item -> item.getFlags().stream().noneMatch(authorFlags::contains));
```

Comment:
Shouldn't this logic be in a service so it can be reused? We probably need the same logic when checking if all criteria items are signed off? During the Snowstorm promotion check.

Reply:
This code assumes that author flags, if set, are always true. The code should check that the value is "true". The set of `authorFlags` could be a filtered set of keys were the value was true.

Reply:
Thanks Kai, I am working on what you've suggested. 

I've moved the logic to a service class and I am currently working on adding the filtering to the promotion workflow.

Reply:
I've moved the logic to `CriteriaItemService`.

I've also worked on adding the flag check to the promotion workflow but I felt like it was _slightly_ different from the viewing PAC workflow.

The viewing PAC workflow now works with a black & white view point, i.e. if the flag is there then show the `CriteriaItem`. If it's not, then don't show it. 

The promotion check workflow is black, white & grey, i.e. if the flag is there then check if it's true/false and act appropriately. If the flag isn't even there, check the CriteriaItem as usual. I felt this prevented the SAC config being stored on both AAG & branch metadata.